pkg/actuators/awsplacementgroup/watch_filters.go

```diff
@@ -0,0 +1,43 @@
+package awsplacementgroup
+
+import (
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	awsmachine "github.com/openshift/machine-api-provider-aws/pkg/actuators/machine"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// machineToAWSPlacementGroup maps a Machine to an AWSPlacementGroup
+// provided that the Machine references one.
+func machineToAWSPlacementGroup(r *AWSPlacementGroupReconciler) func(client.Object) []reconcile.Request {
+	return func(obj client.Object) []reconcile.Request {
+		machine, ok := obj.(*machinev1beta1.Machine)
+		if !ok {
+			// Ignore the Machine if there is any issue asserting the object.
+			return []reconcile.Request{}
```

Comment:
This is a programming error if this assertion fails, so you can panic here with a message along the lines of
```
panic(fmt.Sprintf("expected type *machinev1beta1.Machine, got %T", obj))
```